lib/problems.rb

```diff
@@ -1,13 +1,46 @@
 require_relative './stack.rb'
 
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: O(n)
+# Space Complexity: O(n)
 def balanced(string)
```

Comment:
You have the right idea here with one typo.  See below.

Do consider using a hash, it could make the code more compact.